EPAS 15 new release branch (#2283)
EnterpriseDB/docs: product_docs/docs/epas/15/epas_rel_notes/epas15_2_0_rel_notes.mdx

---
title: "Version 15.2.0"
---

EDB Postgres Advanced Server 15.2.0 includes the following enhancements and bug fixes:

| Type           | Description                                                                                                                          |  Category              |
| -------------- | -------------------------------------------------------------------------------------------------------------------------------------|  --------------------- |
| Upstream merge | Merged with community PostgreSQL 15.2. See the [PostgreSQL 15 Release Notes](https://www.postgresql.org/docs/15/release-15.html) for more information.                           |                        |

> | Feature        | Transparent Data Encryption (TDE) - TDE is an optional feature supported by EDB Postgres Advanced Server with high availability. It encrypts, transparently to the user, any user data stored in the database system. User data includes the actual data stored in tables and other objects, as well as system catalog data such as the names of objects. See [TDE docs](/tde/latest) for more information. | Security |

Reviewer: ```suggestion
| Feature        | Transparent Data Encryption (TDE) encrypts, transparently to the user, any user data stored in the database system. User data includes the actual data stored in tables and other objects, as well as system catalog data such as the names of objects. See [TDE docs](/tde/latest) for more information. | Security |
```

| Enhancement    | EDB Postgres Advanced Server now allows non-superuser to load data using EDB*Loader.     | edbldr |                       

> | Enhancement    | Enable multi-insert support dynamic partition for EDB*Loader and COPY command. |  |

Reviewer: I can't make sense of this one. Can you get someone to explain what this is supposed to be saying? At minimum, I think "support" is misplaced.

Author: done

> | Enhancement    | EDB Postgres Advanced Server now provides the ability to obfuscate the LDAP password in `pg_hba.conf` file. The user can supply a module which will transform the `ldapbindpasswd` in the `pg_hba.conf` file before it is passed to the LDAP server. | Security |

Reviewer: ```suggestion
| Enhancement    | EDB Postgres Advanced Server now provides the ability to obfuscate the LDAP password in the `pg_hba.conf` file. The user can supply a module which will transform the `ldapbindpasswd` value in the `pg_hba.conf` file before it is passed to the LDAP server. | Security |
```

| Enhancement    | Adding OCI dblink configuration file approach to restrict pushdowns. This enhancement adds the infrastructure of the configuration file where you can define the list of operators/functions that can push down to the remote server. It also allows users to add/modify the list as required. |      |
| Enhancement    | MERGE syntax - Adding support for WHERE clause to the UPDATE and INSERT of MERGE command for Oracle compatibility.| Oracle compatibility |

> | Enhancement    | Adding HTP and HTF packages to built-in-packages for Oracle compatibility. | Oracle compatibility |

Reviewer: ```suggestion
| Enhancement    | Adding HTP and HTF packages to built-in packages for Oracle compatibility. | Oracle compatibility |
```

| Enhancement    | Allow INTO clause to accept multiple composite row type targets in SPL. This enhancement allows a SELECT list having a mix of scalar and composite type values that are fetched from a table, to be assigned to corresponding scalar or composite variables (including collection variables) in the SPL code.

> | Enhancement    | Skip IN/OUT/IN OUT modifiers in the USING expression. USING clause in EXECUTE IMMEDIATE does support passing parameters to embedded SPL blocks. These parameters are treatred as IN OUT only, and there is currently no way to specify whether the parameter is IN, OUT or IN OUT. However, to ease migration from Oracle, these modifiers are skipped at the beginning of the expression whenever possible. | Oracle compatibility |

Reviewer: ```suggestion
| Enhancement    | Skip IN/OUT/IN OUT modifiers in the USING expression. USING clause in EXECUTE IMMEDIATE does support passing parameters to embedded SPL blocks. These parameters are treated as IN OUT only, and there is currently no way to specify whether the parameter is IN, OUT, or IN OUT. However, to ease migration from Oracle, these modifiers are skipped at the beginning of the expression whenever possible. | Oracle compatibility |
```

> | Enhancement    | Adding FORMAT_ERROR_STACK() and FORMAT_ERROR_BACKTRACE() functions to DBMS_UTILITY package. These functions are used in a stored procedure, function or package to return the current exception name. These functions are useful for debugging and logging purposes. | |

Reviewer: ```suggestion
| Enhancement    | Adding FORMAT_ERROR_STACK() and FORMAT_ERROR_BACKTRACE() functions to theDBMS_UTILITY package. These functions are used in a stored procedure, function or package to return the current exception name. These functions are useful for debugging and logging purposes. | |
```

> | Enhancement    | Adding Oracle compatible UPDATE..SET ROW syntax support. UPDATE changes the values of the specified columns in all rows that satisfy the condition. Only the columns to modify are mention in the SET clause; columns not to modify explicitly retain their previous values. SET ROW clause enables users to update a target record using a record type variable or row type objects. The condition is that the record or row used should have compatible data types with table's columns in order. | Oracle compatibility |

Reviewer: ```suggestion
| Enhancement    | Adding Oracle compatible UPDATE..SET ROW syntax support. UPDATE changes the values of the specified columns in all rows that satisfy the condition. Only the columns to modify are mentioned in the SET clause; columns not to modify explicitly retain their previous values. The SET ROW clause enables users to update a target record using a record type variable or row type objects. The condition is that the record or row used should have compatible data types with table's columns in order. | Oracle compatibility |
```

> | Enhancement    | EDB Postgres Advanced Server now provides INDEX and NO_INDEX hints for the partitioned table. This optimizer hints applies to inherited index in the partitioned table. The execution plan internally expands to include the corresponding inherited child indexes and applied in later processing. | |

Reviewer: ```suggestion
| Enhancement    | EDB Postgres Advanced Server now provides INDEX and NO_INDEX hints for the partitioned table. The optimizer hints apply to the inherited index in the partitioned table. The execution plan internally expands to include the corresponding inherited child indexes and applies them in later processing. | |
```

| Enhancement    | Adding SQLCODE() and SQLERRM()functions. In an exception handler, the SQLCODE function returns the numeric code of the exception being handled. Outside an exception handler, SQLCODE returns 0. The SQLERRM function returns the error message associated with an SQLCODE variable value. If the error code value is passed to the SQLERRM function it returns an error message associated with the passed error code value, irrespective of the current error raised. |  |
| Enhancement    | Adding TO_MULTI_BYTE() and TO_SINGLE_BYTE() functions. | Oracle compatibility |
| Enhancement    | Adding TO_NCHAR()function. TO_NCHAR() is the wrapper function that casts input to NVARCHAR2. Note that the size of the input is limited to PostgreSQL's supported size limit for that type. | |

> | Enhancement    | Adding TO_DSINTERVAL() function. Convert a character string of CHAR, VARCHAR2, NCHAR, or NVARCHAR2 datatype to an interval datatype. | |

Reviewer: ```suggestion
| Enhancement    | Adding TO_DSINTERVAL() function. Converts a character string of CHAR, VARCHAR2, NCHAR, or NVARCHAR2 datatype to an interval datatype. | |
```
